### ISSUE TITLE
[BUG] Classes that extend class 'o.o.action.support.master.TransportMasterNodeAction' should not have to implement abstract method 'clusterManagerOperation()'

### ISSUE THREAD

**Issue by tlfeng** [opensearch-project/OpenSearch#3683]
**Describe the bug**
Similar issue: https://github.com/opensearch-project/OpenSearch/issues/3663 and https://github.com/opensearch-project/OpenSearch/issues/3688
Introduced by PR https://github.com/opensearch-project/OpenSearch/pull/3617 / commit https://github.com/opensearch-project/OpenSearch/commit/60d7a098492c184e0b2a6fb0492d8c57b7626a54, which aims to resolve issue #3542

The problem:
To deprecate public methods that has "master" in the name, I created an alternative method `clusterManagerOperation(...)` to replace `masterOperation(...)`, which is an abstract method.
Location: https://github.com/opensearch-project/OpenSearch/blob/157e69330891b8133d56ab993b5022f2e8c6ee04/server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java
It makes classes that extend the class [TransportMasterNodeAction](https://github.com/opensearch-project/OpenSearch/blob/157e69330891b8133d56ab993b5022f2e8c6ee04/server/src/main/java/org/opensearch/action/support/master/TransportMasterNodeAction.java) have to implement the newly added abstract method `clusterManagerOperation(...)`, and it breaks the backwards compatibility.

The solution is to revert the changes to remove the abstract method `clusterManagerOperation(...)`.

**To Reproduce**
Build [k-NN plugin](https://github.com/opensearch-project/k-NN) with OpenSearch 2.1.0 (the code at https://github.com/opensearch-project/OpenSearch/tree/304d830275fdbe5f8095a7285ca66266987c727a)

`java.lang.AbstractMethodError: Missing implementation of resolved method 'abstract void clusterManagerOperation(org.opensearch.action.support.clustermanager.ClusterManagerNodeRequest, org.opensearch.cluster.ClusterState, org.opensearch.action.ActionListener)' of abstract class org.opensearch.action.support.clustermanager.TransportClusterManagerNodeAction.
»  	at org.opensearch.action.support.clustermanager.TransportClusterManagerNodeAction.clusterManagerOperation(TransportClusterManagerNodeAction.java:135)
»  	at org.opensearch.action.support.clustermanager.TransportClusterManagerNodeAction$AsyncSingleAction.lambda$doStart$3(TransportClusterManagerNodeAction.java:223)`

**Expected behavior**
k-NN plugin should successfully be built with OpenSearch 2.1.0

**Additional context**

**Comment by tlfeng** [comment]
The temporary solution has been applied to `2.1` branch by the PR #3682

**Comment by tlfeng** [comment]
Close the issue. The `masterOperation()` method name have been restored in all branches.
PR https://github.com/opensearch-project/OpenSearch/pull/3681 for `main` branch, and PR https://github.com/opensearch-project/OpenSearch/pull/3714 for `2.x` branch.

**Comment by tlfeng** [comment]
I have got a solution from @andrross to properly deprecate the abstract method 'masterOperation()'.
```
@Deprecated("override doSomethingWithClusterManager instead")
void doSomethingWithMaster() {
    throw new UnsupportedOperationException("Must be overridden");
}

void doSomethingWithClusterManager() {
    doSomethingWithMaster();
}
```

The below is wrong because it will break any child classes.
```
@Deprecated("override doSomethingWithClusterManager instead")
abstract void doSomethingWithMaster();

abstract void doSomethingWithClusterManager();
```

**Comment by tlfeng** [comment]
The final solution in PR https://github.com/opensearch-project/OpenSearch/pull/4032 has been merged, and from version 2.2.0, developers can use `clusterManagerOperation()` to replace `masterOperation()`.
For more detail: https://github.com/opensearch-project/OpenSearch/issues/3542#issuecomment-1200078089

**Comment by tlfeng** [comment]
As mentioned above, the 4 classes `AcknowledgedRequest`, `AcknowledgedRequestBuilder`, `AcknowledgedResponse`, `ShardsAcknowledgedResponse` in package `org.opensearch.action.support.master` will not be deprecated in version `2.x`, and will be moved to package `org.opensearch.action.support.clustermanager` directly in a future major version as a breaking change.
